qiskit/algorithms/eigensolvers/ssvqe.py

```diff
@@ -0,0 +1,575 @@
+                    [np.asarray(Statevector(state)) for state in initial_states]
+                )
+                if not np.isclose(
+                    stacked_states_array.transpose() @ stacked_states_array, np.eye(self.k)
+                ):
+                    raise AlgorithmError(
+                        "The set of initial states provided is not mutually orthogonal."
+                    )
+
+        for initial_state in initial_states:
+            initial_state = self._check_circuit_num_qubits(
+                operator=operator, circuit=initial_state, circuit_type="initial state"
+            )
+
+        return initial_states
+
+    def _check_weight_vector(self, weight_vector: Sequence[float]) -> Sequence[float]:
+        """Check that the number of weights matches the number of states."""
+        if weight_vector is None:
+            weight_vector = [self.k - n for n in range(self.k)]
+        elif len(weight_vector) != self.k:
+            raise AlgorithmError(
+                "The number of weights provided does not match the number of states."
+            )
+
+        return weight_vector
+
+    def _eval_aux_ops(
+        self,
+        ansatz: QuantumCircuit,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp],
+    ) -> ListOrDict[tuple(complex, complex)]:
+        """Compute auxiliary operator eigenvalues."""
+
+        if isinstance(aux_operators, dict):
+            aux_ops = list(aux_operators.values())
+        else:
+            aux_ops = aux_operators
+
+        num_aux_ops = len(aux_ops)
+        aux_job = self.estimator.run([ansatz] * num_aux_ops, aux_ops)
+        aux_values = aux_job.result().values
+        aux_values = list(zip(aux_values, [0] * len(aux_values)))
+
+        if isinstance(aux_operators, dict):
+            aux_values = dict(zip(aux_operators.keys(), aux_values))
+
+        return aux_values
+
+    def _build_ssvqe_result(
+        self,
+        optimizer_result: OptimizerResult,
+        aux_operators_evaluated: ListOrDict[tuple[complex, tuple[complex, int]]],
+        optimizer_time: float,
+        operator: BaseOperator | PauliSumOp,
+        initialized_ansatz_list: Sequence[QuantumCircuit],
+    ) -> SSVQEResult:
+        result = SSVQEResult()
+        result.eigenvalues = (
+            self.estimator.run(
+                initialized_ansatz_list, [operator] * self.k, [optimizer_result.x] * self.k
+            )
+            .result()
+            .values
+        )
+        result.cost_function_evals = optimizer_result.nfev
+        result.optimal_point = optimizer_result.x
+        result.optimal_parameters = dict(zip(self.ansatz.parameters, optimizer_result.x))
+        result.optimal_value = optimizer_result.fun
+        result.optimizer_time = optimizer_time
+        result.aux_operators_evaluated = aux_operators_evaluated
+        result.optimizer_result = optimizer_result
+
+        return result
+
+
+class SSVQEResult(VariationalResult, EigensolverResult):
```

Comment:
How about we move this class to a separate file?

Reply:
I put `SSVQEResult` in this file to keep it consistent with `VQE` and `VQD` , which also keep their respective Result class in the same file, but if there is recent code refactoring that I have missed, let me know.